### PR TITLE
NEW Enable auto update PDF on update note

### PR DESCRIPTION
**Pull request by all3kcis**
Generate order/invoice PDF on update public note

### CODE REVIEW

File: htdocs/core/class/commonobject.class.php [hunk 1]
```diff
@@ -2511,6 +2511,27 @@ function update_note($note,$suffix='')
 				$this->note = $note;      // deprecated
 				$this->note_private = $note;
 			}
+			if($suffix != '_private' && in_array($this->table_element, array('commande_fournisseur', 'commande', 'propal', 'facture_fourn', 'facture' ))){
```
Comment (eldy) [inline htdocs/core/class/commonobject.class.php]: The "update_note" method is a CRUD method to affect the database values. I would prefer to see code to build PDF that is not related with DAO principle after the call of update_note

**Comment by all3kcis** [comment]
I can transfer code into "htdocs/core/actions_setnotes.inc.php", Ok ?

**Comment by eldy** [comment]
Yes, good idea.

File: htdocs/core/actions_setnotes.inc.php [hunk 2]
```diff
@@ -32,8 +32,32 @@
...
+			if ($conf->global->MAIN_MULTILANGS && empty($newlang))	$newlang = $object->thirdparty->default_lang;
+			if (! empty($newlang)) {
+				$outputlangs = new Translate("", $conf);
+				$outputlangs->setDefaultLang($newlang);
+			}
+			$model=$object->modelpdf;
+			$hidedetails = (GETPOST('hidedetails','int') ? GETPOST('hidedetails','int') : (! empty($conf->global->MAIN_GENERATE_DOCUMENTS_HIDE_DETAILS) ? 1 : 0));
+			$hidedesc = (GETPOST('hidedesc','int') ? GETPOST('hidedesc','int') : (! empty($conf->global->MAIN_GENERATE_DOCUMENTS_HIDE_DESC) ?  1 : 0));
+			$hideref = (GETPOST('hideref','int') ? GETPOST('hideref','int') : (! empty($conf->global->MAIN_GENERATE_DOCUMENTS_HIDE_REF) ? 1 : 0));
+			require_once DOL_DOCUMENT_ROOT.'/fourn/class/fournisseur.product.class.php';
```
Comment (eldy) [inline htdocs/core/actions_setnotes.inc.php]: Is this require_once necessary ?

Comment (all3kcis) [reply htdocs/core/actions_setnotes.inc.php]: Of course

Comment (eldy) [reply htdocs/core/actions_setnotes.inc.php]: I don't understand why.
There is no dedicated code to supplier product added by the PR, so why adding this ? The require is surely at the wrong place.
